x-pack/plugins/security_solution/public/management/components/endpoint_responder/get_running_processes_action.tsx

```diff
@@ -0,0 +1,198 @@
+
+import React, { memo, useEffect, useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ActionDetails, RunningProcessesEntry } from '../../../../common/endpoint/types';
+import { useGetActionDetails } from '../../hooks/endpoint/use_get_action_details';
+import { EndpointCommandDefinitionMeta } from './types';
+import { CommandExecutionComponentProps } from '../console/types';
+import { useSendGetEndpointRunningProcessesRequest } from '../../hooks/endpoint/use_send_get_endpoint_running_processes_request';
+
+// @ts-expect-error TS2769
+const StyledEuiBasicTable = styled(EuiBasicTable)`
+  table {
+    background-color: ${({ theme: { eui } }) => eui.euiPageBackgroundColor};
+  }
+  .euiTableHeaderCell {
+    border-bottom: ${(props) => props.theme.eui.euiBorderThin};
+    .euiTableCellContent__text {
+      font-weight: 400;
+    }
+  }
+  .euiTableRow {
+    &:hover {
+      background-color: white !important;
+    }
+    .euiTableRowCell {
+      border-top: none !important;
+      border-bottom: none !important;
+    }
+  }
+`;
+
+export const GetRunningProcessesActionResult = memo<
+  CommandExecutionComponentProps<
+    { comment?: string },
+    {
+      actionId?: string;
+      actionRequestSent?: boolean;
+      completedActionDetails?: ActionDetails<RunningProcessesEntry>;
+    },
+    EndpointCommandDefinitionMeta
+  >
+>(({ command, setStore, store, status, setStatus, ResultComponent }) => {
+  const endpointId = command.commandDefinition?.meta?.endpointId;
+  const { actionId, completedActionDetails } = store;
+
+  const isPending = status === 'pending';
+  const actionRequestSent = Boolean(store.actionRequestSent);
+
+  const getRunningProcessesApi = useSendGetEndpointRunningProcessesRequest();
+
+  const { data: actionDetails } = useGetActionDetails<RunningProcessesEntry>(actionId ?? '-', {
+    enabled: Boolean(actionId) && isPending,
+    refetchInterval: isPending ? 3000 : false,
+  });
+
+  // Send get running processes request if not yet done
+  useEffect(() => {
+    if (!actionRequestSent && endpointId) {
+      getRunningProcessesApi.mutate({
+        endpoint_ids: [endpointId],
+        comment: command.args.args?.comment?.[0],
+      });
+
+      setStore((prevState) => {
+        return { ...prevState, actionRequestSent: true };
+      });
+    }
+  }, [actionRequestSent, command.args.args?.comment, endpointId, getRunningProcessesApi, setStore]);
+
+  // If get running processes request was created, store the action id if necessary
+  useEffect(() => {
+    if (getRunningProcessesApi.isSuccess && actionId !== getRunningProcessesApi.data.data.id) {
+      setStore((prevState) => {
+        return { ...prevState, actionId: getRunningProcessesApi.data.data.id };
+      });
+    } else {
+      setStore((prevState) => {
+        if (prevState.completedActionDetails) {
```

Comment:
Just a heads up that this won't work on an API error since there may not be an action created.  We likely need to introduce a section just for API errors.  `completedActionDetails` may never be previously populated since the API fails (think of a `404` or `500`), so we would need to just fill in an API error and make display that.